src/pgwire/src/codec.rs

```diff
@@ -476,7 +506,220 @@ fn decode_terminate(mut _buf: Cursor) -> Result<FrontendMessage, io::Error> {
     Ok(FrontendMessage::Terminate)
 }
 
-fn decode_password(mut buf: Cursor) -> Result<FrontendMessage, io::Error> {
+fn decode_auth(mut buf: Cursor) -> Result<FrontendMessage, io::Error> {
+    let mut value = Vec::new();
+    while let Ok(b) = buf.read_byte() {
+        value.push(b);
+    }
+    Ok(FrontendMessage::RawAuthentication(value))
+}
+
+fn expect(buf: &mut Cursor, expected: &[u8]) -> Result<(), io::Error> {
+    for i in 0..expected.len() {
+        if buf.read_byte()? != expected[i] {
+            return Err(input_err(format!(
+                "Invalid SASL initial response: expected '{}'",
+                std::str::from_utf8(expected).unwrap_or("invalid UTF-8")
+            )));
+        }
+    }
+    Ok(())
+}
+
+fn read_until_comma(buf: &mut Cursor) -> Result<Vec<u8>, io::Error> {
+    let mut v = Vec::new();
+    while let Ok(b) = buf.peek_byte() {
+        if b == b',' {
+            break;
+        }
+        v.push(buf.read_byte()?);
+    }
+    Ok(v)
+}
+
+//   extensions = attr-val *("," attr-val)
+//                     ;; All extensions are optional,
+//                     ;; i.e., unrecognized attributes
+//                     ;; not defined in this document
+//                     ;; MUST be ignored.
+//   reserved-mext  = "m=" 1*(value-char)
+//                     ;; Reserved for signaling mandatory extensions.
+//                     ;; The exact syntax will be defined in
+//                     ;; the future.
+//   gs2-cbind-flag  = ("p=" cb-name) / "n" / "y"
+//                     ;; "n" -> client doesn't support channel binding.
+//                     ;; "y" -> client does support channel binding
+//                     ;;        but thinks the server does not.
+//                     ;; "p" -> client requires channel binding.
+//                     ;; The selected channel binding follows "p=".
+//
+//   gs2-header      = gs2-cbind-flag "," [ authzid ] ","
+//                     ;; GS2 header for SCRAM
+//                     ;; (the actual GS2 header includes an optional
+//                     ;; flag to indicate that the GSS mechanism is not
+//                     ;; "standard", but since SCRAM is "standard", we
+//                     ;; don't include that flag).
+//   client-first-message-bare =
+//                     [reserved-mext ","]
+//                     username "," nonce ["," extensions]
+//
+//   client-first-message =
+//                     gs2-header client-first-message-bare
+pub fn decode_sasl_client_first_message(mut buf: Cursor) -> Result<SASLInitialResponse, io::Error> {
+    // 1) GS2 cbind flag
+    let cbind_flag = match buf.read_byte()? {
```

Comment:
At least from this code review, it seems like we don't support channel binding? It also seems like we don't really do anything with the gs2_header. I feel like at the very least, we should error out on `p` given `p` says the channel requires channel binding

Reply:
Also maybe making it explicitly clear that we don't support channel binding, maybe in the ChannelBinding enum?